### PR TITLE
feat: show ReplyTo in widget UI

### PR DESCRIPTION
**Pull request by scmmishra**
| Dark Mode | Light Mode |
|:---:|:---:|
| dark | light |

### CODE REVIEW

File: app/javascript/widget/components/ReplyToChip.vue [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    },
+  },
+};
+</script>
+
+<style lang="scss">
+.reply-to-truncate > p {
+  @apply overflow-hidden truncate max-w-[8rem];
+}
+</style>
```
Comment (iamsivin) [inline app/javascript/widget/components/ReplyToChip.vue:57]: @scmmishra Are we using these styles anywhere?

File: app/javascript/widget/components/FooterReplyTo.vue [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<template>
+  <div
+    class="mb-2.5 rounded-[7px] dark:bg-slate-900 dark:text-slate-100 bg-slate-100 px-2 py-1.5 text-sm text-slate-700 flex items-center gap-2"
+  >
+    <div class="items-center flex-grow truncate">
```
Comment (iamsivin) [inline app/javascript/widget/components/FooterReplyTo.vue:5]: Let us use it from the Tailwind itself.

```suggestion
    <div class="items-center flex-grow overflow-hidden whitespace-nowrap text-ellipsis">
```

File: app/javascript/widget/components/ChatFooter.vue [hunk 3]
```diff
@@ -1,15 +1,25 @@
...
+        hasReplyTo,
+    }"
   >
+    <footer-reply-to
+      v-if="hasReplyTo"
+      :in-reply-to="inReplyTo"
+      @dismiss="inReplyTo = null"
+    />
     <chat-input-wrap
+      class="bg-white shadow-sm"
```
Comment (iamsivin) [inline app/javascript/widget/components/ChatFooter.vue:17]: @scmmishra Do we need this `bg-white` here? I think the component already has.

File: app/javascript/widget/components/ChatFooter.vue [hunk 4]
```diff
@@ -1,15 +1,25 @@
...
       :on-send-message="handleSendMessage"
       :on-send-attachment="handleSendAttachment"
     />
   </footer>
-  <div v-else>
+  <div v-else class="bg-white shadow-sm">
```
Comment (iamsivin) [inline app/javascript/widget/components/ChatFooter.vue:22]: The background for the dark mode is missing here.

**Review by iamsivin — COMMENTED** [review]
@scmmishra Added some comments. Please take a look.

File: app/javascript/widget/components/ReplyToChip.vue (hunk 1 above)
Comment (scmmishra) [reply app/javascript/widget/components/ReplyToChip.vue:57]: Nopes, maybe I can scope it with v-deep. Let me try

File: app/javascript/widget/components/FooterReplyTo.vue (hunk 2 above)
Comment (scmmishra) [reply app/javascript/widget/components/FooterReplyTo.vue:5]: https://tailwindcss.com/docs/text-overflow#truncate

`truncate` is a standard class

Comment (iamsivin) [reply app/javascript/widget/components/FooterReplyTo.vue:5]: Fine

File: app/javascript/widget/components/ChatFooter.vue (hunk 3 above)
Comment (scmmishra) [reply app/javascript/widget/components/ChatFooter.vue:17]: Yeah, we can remove it

```suggestion
      class="shadow-sm"
```

File: app/javascript/widget/components/ReplyToChip.vue [hunk 5]
```diff
@@ -0,0 +1,57 @@
+<template>
+  <button
+    class="px-1.5 py-0.5 rounded-md text-slate-500 bg-slate-50 dark:bg-slate-900 opacity-60 hover:opacity-100 cursor-pointer flex items-center gap-1.5"
```
[the remark this answers is not in the document]
Comment (scmmishra) [reply app/javascript/widget/components/ReplyToChip.vue:3]: Noted. Will fix

File: app/javascript/widget/components/ChatFooter.vue (hunk 4 above)
Comment (scmmishra) [reply app/javascript/widget/components/ChatFooter.vue]: Fixed this, seems like this is not necessary

File: app/javascript/widget/components/ReplyToChip.vue (hunk 5 above)
[the remark this answers is not in the document]
Comment (scmmishra) [reply app/javascript/widget/components/ReplyToChip.vue:3]: Done

**Comment by muhsin-k** [comment]
The reply icon is currently positioned on the left side for text messages and at the top for attachments. Is there a specific reason for this placement? Can we make the placement consistent across both text messages and attachments?

CleanShot 2023-10-25 at 14 30 13

**Comment by scmmishra** [comment]
> @scmmishra The reply icon is currently positioned on the left side for text messages and at the top for attachments. Is there a specific reason for this placement? Can we make the placement consistent across both text messages and attachments?
> 
> CleanShot 2023-10-25 at 14 30 13

It's unintentional, let me fix that

**Comment by scmmishra** [comment]
@muhsin-k fixed it

**Review by muhsin-k — APPROVED** [review]

**Pull request merged**